DEVELOPMENT.md

```diff
@@ -10,6 +10,104 @@ This document outlines the process for setting up the development environment fo
 
 ## Setting up the Environment
 
+Decide first if you'd like a full buildsystem (on a VM) or a containerized dev environment.
+
+### VM as buildsystem
+Uses a Ubuntu 24.04 as base to run `chef` to setup all dependencies.
+The initial compilation is CPU intense and `16vcpu` are recommended.
+This was tested on GCP: a balanced disk of 500 GB and a VM type that supports nested virtualization should be chosen, as of writing (May 2025)
+`n2-standard-16` works well. 
+
+> [!Warning]
+>  The first build takes several hours and at least 160 Gb of space
+
+Turn on nested virtualization and avoid the use of `spot` VMs for the first build to avoid the very long first
+build interrupting. If you create the VMs as templates from an image, you can later switch to more cost-effective `spot` instances.
```

Comment:
Can we specify that nested virtualization is only needed if you intend to develop against Minikube?

Also I think the details about spot instances is a little niche as it's possible that people are running non cloud VMs. Could we remove those details?

Reply:
I removed and added explanation